bridge/client/app/_components/ktb-create-secret-form/ktb-create-secret-form.component.ts

```diff
@@ -1,66 +1,104 @@
+import { Subject } from 'rxjs';
 
 @Component({
   selector: 'ktb-secrets-view',
   templateUrl: './ktb-create-secret-form.component.html',
-  styleUrls: [],
+  styleUrls: ['./ktb-create-secret-form.component.scss'],
 })
-export class KtbCreateSecretFormComponent {
-  public isLoading = false;
-
+export class KtbCreateSecretFormComponent implements OnDestroy {
   private secretNamePattern = '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\\\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*';
   private secretKeyPattern = '[-._a-zA-Z0-9]+';
+  private _isLoading = false;
+  private unsubscribe$ = new Subject<void>();
+  public FormGroupClass = FormGroup;
+  public scopeControl = new FormControl(undefined, [Validators.required]);
+  public nameControl = new FormControl('', [
+    Validators.required,
+    Validators.pattern(this.secretNamePattern),
+    Validators.maxLength(253),
+  ]);
+  public dataControl = this.fb.array([
+    this.fb.group({
+      key: ['', [Validators.required, Validators.pattern(this.secretKeyPattern), Validators.maxLength(253)]],
+      value: ['', [Validators.required]],
+    }),
+  ]);
 
-  public scopes: SecretScope[] = Object.values(SecretScope);
-
+  public isUpdating = false;
+  public scopes?: string[];
   public createSecretForm = this.fb.group({
-    name: ['', [Validators.required, Validators.pattern(this.secretNamePattern), Validators.maxLength(253)]],
-    scope: [this.scopes[0], [Validators.required]],
-    data: this.fb.array([
-      this.fb.group({
-        key: ['', [Validators.required, Validators.pattern(this.secretKeyPattern), Validators.maxLength(253)]],
-        value: ['', [Validators.required]],
-      }),
-    ]),
+    name: this.nameControl,
+    scope: this.scopeControl,
+    data: this.dataControl,
   });
 
+  public get isLoading(): boolean {
+    return this._isLoading;
+  }
+  public set isLoading(isLoading: boolean) {
+    this._isLoading = isLoading;
+    if (isLoading) {
+      this.scopeControl.disable();
+    } else {
+      this.scopeControl.enable();
+    }
+  }
+
   constructor(
     private dataService: DataService,
     private router: Router,
     private route: ActivatedRoute,
     private fb: FormBuilder,
-    private notificationService: NotificationsService
-  ) {}
-
-  get data(): FormArray | null {
-    return this.createSecretForm.get('data') as FormArray;
+    private notificationService: NotificationsService,
+    @Inject(POLLING_INTERVAL_MILLIS) private initialDelayMillis: number
+  ) {
+    this.getSecretScopes();
   }
 
-  get dataControls(): FormGroup[] {
-    return (this.data?.controls as FormGroup[]) || [];
+  private getSecretScopes(): void {
+    AppUtils.createTimer(0, this.initialDelayMillis) // interval if it fails to fetch scopes
+      .pipe(
+        takeWhile(() => !this.scopes),
+        tap(() => {
+          this.isLoading = true;
+        }),
+        switchMap(() => this.dataService.getSecretScopes()),
+        takeUntil(this.unsubscribe$)
+      )
```

Comment:
How often will it retry in case of a failure? 
Some inline docs would be nice here

Reply:
At the moment there isn't a limit and the retry happens until the user leaves the page

Reply:
I think a limit would make sense, maybe max. 3-5 times, max. 120 seconds (if it doesn't resolve until then, it's probably broken and needs manual intervention by the admin)

Reply:
I just remembered we actually have a retry mechanism already in place, but this one immediately tries again (up to 3 times). I guess we can remove the implementation here.